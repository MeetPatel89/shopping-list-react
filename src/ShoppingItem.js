import React from 'react';
import './ShoppingItem.css';

export default class ShoppingItem extends React.Component {

    static defaultProps ={
        item: {}
    }

    render() {

        

        return (
            <li>
                <h2 style={{textDecoration: this.props.item.checked && "line-through"}}>{this.props.item.name}</h2>
                <button type="button" onClick={() => this.props.onCheckItem(this.props.item.name)}>Check</button>
                <button type="button" onClick={() => this.props.onDeleteItem(this.props.item.name)}>Delete</button>
            </li>
        )
    }
}