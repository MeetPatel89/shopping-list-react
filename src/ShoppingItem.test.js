import React from 'react';
import ReactDOM from 'react-dom';
import ShoppingItem from './ShoppingItem';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

describe('ShoppingItem Component', () => {

    test('Renders component without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<ShoppingItem/>, div);
        ReactDOM.unmountComponentAtNode(div);
    })

    test('renders empty given no item', () => {
        const wrapper = shallow(<ShoppingItem/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    test('renders the item when supplied', () => {
        const testItem = {name: 'test-item', checked: false}
        const wrapper = shallow(<ShoppingItem item={testItem}/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    test('renders checked item when checked property is true', () => {
        const testItem = {name: 'test-item', checked: true}
        const wrapper = shallow(<ShoppingItem item={testItem}/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})