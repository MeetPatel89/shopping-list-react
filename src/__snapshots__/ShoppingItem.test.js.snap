// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShoppingItem Component renders checked item when checked property is true 1`] = `
<li>
  <h2
    style={
      Object {
        "textDecoration": "line-through",
      }
    }
  >
    test-item
  </h2>
  <button
    type="button"
  >
    Check
  </button>
  <button
    type="button"
  >
    Delete
  </button>
</li>
`;

exports[`ShoppingItem Component renders empty given no item 1`] = `
<li>
  <h2
    style={
      Object {
        "textDecoration": undefined,
      }
    }
  />
  <button
    type="button"
  >
    Check
  </button>
  <button
    type="button"
  >
    Delete
  </button>
</li>
`;

exports[`ShoppingItem Component renders the item when supplied 1`] = `
<li>
  <h2
    style={
      Object {
        "textDecoration": false,
      }
    }
  >
    test-item
  </h2>
  <button
    type="button"
  >
    Check
  </button>
  <button
    type="button"
  >
    Delete
  </button>
</li>
`;
